vars/cloudFoundryDeploy.groovy

```diff
@@ -0,0 +1,161 @@
+            // set default mtar path
+            config = new ConfigurationHelper(config)
+                .addIfEmpty('mtaPath', config.mtaPath?:findMtar())
+                .use()
+
+            dockerExecute(dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent) {
+                deployMta(config)
+            }
+            return
+        }
+
+        if (config.deployTool == 'cf_native') {
+            def smokeTest = ''
+
+            if (config.smokeTestScript == 'blueGreenCheck.sh') {
+                writeFile file: config.smokeTestScript, text: libraryResource(config.smokeTestScript)
+            } else {
+                utils.unstash 'pipelineConfigAndTests'
+            }
+            smokeTest = '--smoke-test $(pwd)/' + config.smokeTestScript
+            sh "chmod +x ${config.smokeTestScript}"
+
+            echo "[${STEP_NAME}] CF native deployment (${config.deployType}) with cfAppName=${config.cloudFoundry.appName}, cfManifest=${config.cloudFoundry.manifest}, smokeTestScript=${config.smokeTestScript}"
+
+            dockerExecute (dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent, dockerEnvVars: [CF_HOME:"${config.dockerWorkspace}", CF_PLUGIN_HOME:"${config.dockerWorkspace}", STATUS_CODE: "${config.smokeTestStatusCode}"]) {
+                deployCfNative(config)
+            }
+
+            return
+        }
+    }
+}
+
+def findMtar(){
+    def mtarPath = ''
+    def mtarFiles = findFiles(glob: '**/target/*.mtar')
+
+    if(mtarFiles.length > 1){
+        error 'Found multiple *.mtar files, please specify file via mtaPath parameter! ${mtarFiles}'
+    }
+    if(mtarFiles.length == 1){
+        return mtarFiles[0].path
+    }
+    error 'No *.mtar file found!'
+}
+
+def deployCfNative (config) {
+    withCredentials([usernamePassword(
+        credentialsId: config.cloudFoundry.credentialsId,
+        passwordVariable: 'password',
+        usernameVariable: 'username'
+    )]) {
+        def deployCommand = 'push'
+        if (config.deployType == 'blue-green') {
+            deployCommand = 'blue-green-deploy'
+        } else {
+            config.smokeTest = ''
+        }
+
+        // check if appName is available
+        if (config.cloudFoundry.appName == null || config.cloudFoundry.appName == '') {
+            if (fileExists(config.cloudFoundry.manifest)) {
+                def manifest = readYaml file: config.cloudFoundry.manifest
+                if (!manifest || !manifest.applications || !manifest.applications[0].name)
+                    error "[${STEP_NAME}] ERROR: No appName available in manifest ${config.cloudFoundry.manifest}."
+
+            } else {
+                error "[${STEP_NAME}] ERROR: No manifest file ${config.cloudFoundry.manifest} found."
+            }
+        }
+
+        sh """#!/bin/bash
+            set +x
+            export HOME=/home/piper
```

Comment:
Should this value dependent on the configuration instead of being a fixed one?

Reply:
good point! Done ...